src/error.rs

```diff
@@ -89,6 +89,13 @@ pub trait Error<'a, I: Input<'a>>: Sized {
 #[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Copy, Clone)]
 pub struct EmptyErr(());
 
+impl EmptyErr {
+    /// TODO
```

Comment:
Feel free to make me actually document things in PRs as I go - I have a habit of just making changes then going back later, but that's not a great habit for a library this big probably :P

Reply:
I was fine with it provided there is a `TODO`: throw them everywhere you can, they're easy to search for.

Reply:
Maybe it's better to just have `EmptyErr` implement `Default`, by the way?

Reply:
That's probably a good idea, yeah